pkg/handler/collector/github/github.go

```diff
@@ -0,0 +1,159 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+)
+
+const (
+	CollectorGitHubDocument = "GitHubCollector"
+)
+
+type githubDocumentCollector struct {
+	poll     bool
+	interval time.Duration
+	token    string
+	owner    string
+	repo     string
+	tag      string
+	tagList  []string
+}
+
+func NewGitHubDocumentCollector(ctx context.Context, poll bool, interval time.Duration, logger *zap.SugaredLogger, token string, owner string, repo string, tag string, tagList []string) *githubDocumentCollector {
+	return &githubDocumentCollector{
+		poll:     poll,
+		interval: interval,
+		token:    token,
+		owner:    owner,
+		repo:     repo,
+		tag:      tag,
+		tagList:  tagList,
+	}
+}
+
+func (g *githubDocumentCollector) RetrieveArtifacts(ctx context.Context, docChannel chan<- *processor.Document) error {
+	logger := logging.FromContext(ctx)
+	if g.poll {
+		for {
+			if ctx.Err() != nil {
+				return nil
+			}
+			err := g.fetchAssets(ctx, logger, docChannel)
```

Comment:
Need to check that a tag is not passed in. As the poll should always be querying for the latest release.